pkg/monitor/agent_test.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitor
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	cgtesting "k8s.io/client-go/testing"
+
+	"antrea.io/antrea/pkg/agent/config"
+	interfacestoretest "antrea.io/antrea/pkg/agent/interfacestore/testing"
+	openflowtest "antrea.io/antrea/pkg/agent/openflow/testing"
+	"antrea.io/antrea/pkg/agent/querier"
+	"antrea.io/antrea/pkg/apis/crd/v1beta1"
+	fakeclientset "antrea.io/antrea/pkg/client/clientset/versioned/fake"
+	binding "antrea.io/antrea/pkg/ovs/openflow"
+	ovsconfigtest "antrea.io/antrea/pkg/ovs/ovsconfig/testing"
+	queriertest "antrea.io/antrea/pkg/querier/testing"
+)
+
+const ovsVersion = "2.10.0"
+
+func TestSyncAgentCRD(t *testing.T) {
+	existingCRD := &v1beta1.AntreaAgentInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "testAgentCRD",
+		},
+		NetworkPolicyControllerInfo: v1beta1.NetworkPolicyControllerInfo{
+			NetworkPolicyNum: 1,
+		},
+		APIPort: 0,
+	}
+	partiallyUpdatedCRD := &v1beta1.AntreaAgentInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: crdName,
+		},
+		NetworkPolicyControllerInfo: v1beta1.NetworkPolicyControllerInfo{
+			NetworkPolicyNum: 0,
+		},
+		APIPort: 0,
+	}
+	entirelyUpdatedCRD := &v1beta1.AntreaAgentInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: crdName,
+		},
+		NetworkPolicyControllerInfo: v1beta1.NetworkPolicyControllerInfo{
+			NetworkPolicyNum: 0,
+		},
+		APIPort: 10349,
+	}
+	t.Run("partial update-success", func(t *testing.T) {
+		clientset := fakeclientset.NewSimpleClientset(existingCRD)
+		monitor := newAgentMonitor(clientset, t)
+		monitor.agentCRD = existingCRD
+		monitor.querier.GetAgentInfo(partiallyUpdatedCRD, true)
+		monitor.syncAgentCRD()
+		crd, err := monitor.client.CrdV1beta1().AntreaAgentInfos().Get(context.TODO(), "testAgentCRD", metav1.GetOptions{})
```

Comment:
I suggest declaring `ctx := context.Background()` at the top of the test and using that context for all your sub-tests. Same for controller tests.

Reply:
fixed